lua/orgmode/colors/markup_highlighter.lua

```diff
@@ -173,15 +221,38 @@ local function get_matches(bufnr, first_line, last_line)
   for _, match, _ in query:iter_matches(root, bufnr, first_line, last_line) do
     for _, node in pairs(match) do
       local char = node:type()
-      local range = ts_utils.node_to_lsp_range(node)
-      local linenr = tostring(range.start.line)
-      taken_locations[linenr] = taken_locations[linenr] or {}
-      if not taken_locations[linenr][range.start.character] then
-        table.insert(ranges, {
-          type = char,
-          range = range,
-        })
-        taken_locations[linenr][range.start.character] = true
+      -- markups that start with a backslash \\ will be followed by another
```

Comment:
This for loop still has "char" specific logic which does not belong here. It should just populate the ranges, and next loop should be in charge on any char specific logic (See how links are handled form line 305 - 315. Also, there should be a separate result entry for the math, like we have for the markup (`result`) and links (`link_result`).

Reply:
I think it’s possible to move the char logic further down. Now, I understand the purpose of this loop better. I will fix that.